snippets/buy-buttons.liquid

```diff
@@ -42,22 +42,22 @@
           class="product-variant-id"
         >
         <div class="product-form__buttons">
+          {%- liquid
+            if product.selected_or_first_available_variant.quantity_rule.min > product.selected_or_first_available_variant.inventory_quantity and product.selected_or_first_available_variant.inventory_policy != 'continue'
+              assign quantity_rule_atc_disabled = true
```

Comment:
The logic is tied to the quantity rules and disabling the Add To Cart/Buy buttons based on its logic. 
The quantity rule is always applied but we're just checking to make sure whether it should "disable" the buy buttons or not. 
What do you think of `quantity_rule_soldout` ? Because it's basically what I'm checking. Whether we should set the buttons to a sold out state or not depending on inventory settings on the product/variant.

Reply:
I'm ok with `quantity_rule_soldout`